include/deal.II/grid/tensor_product_manifold.h

```diff
@@ -205,9 +196,18 @@ TensorProductManifold<dim,
       internal::TensorProductManifoldImplementation::concat(
         manifold_A.get_periodicity(),
         manifold_B.get_periodicity()))
-  , manifold_A(&manifold_A)
-  , manifold_B(&manifold_B)
-{}
+{
+  std::shared_ptr<Manifold<dim_A, spacedim_A>> tmpA(
+    std::move(manifold_A.clone()));
+  std::shared_ptr<Manifold<dim_B, spacedim_B>> tmpB(
+    std::move(manifold_B.clone()));
+  this->manifold_A =
+    std::static_pointer_cast<ChartManifold<dim_A, spacedim_A, chartdim_A>>(
+      tmpA);
```

Comment:
you can avoid the temporary here as well if you use `Utilities::unique_dynamic_cast` right on the return type of the `clone()` calls above.

Reply:
Actually, if you use `Utilities::unique_dynamic_cast`, then the `shared_ptr` above can become `unique_ptr`.